R/toolCalcEnergyProj.R

```diff
@@ -0,0 +1,142 @@
+#' @param subset String, designating the GDP scenarios to use
+#' @seealso [convertNewClimate()]
+toolCalcEnergyProj <- function(subtype, subset, years = seq(2020, 2050, 5)) {
+  # This function projects FE demand and SE electricity generation on country-level to future years based on a FE and GDP trend approach:
+  #
+  # Energy(targetYear) = Energy(2020) * [ FE(REMINDregion,targetYear) / FE(REMINDregion,2020)
+  #                                         + ( GDP(country,targetYear) / GDP(country,2020) - GDP(REMINDregion,targetYear) / GDP(REMINDregion,2020) ) ]
+  #
+  # if subtype = "FE" -> Energy is total final energy, FE is total final energy consumption
+  # if subtype = "SE|Electricity" -> Energy is total secondary energy electricity generation, FE is FE electricity consumption
+  # This means that, for example, future country-level FE demand is projected by multiplying historical FE demand with the sum of
+  # a) the FE demand trend from the REMIND region of this country and
+  # b) the difference in GDP between the country and the REMIND region.
+
+  # The projection applies four steps:
+  # 1. get historical energy data for 2020
+  # 2. calculate trend of FE in REMIND region based on EDGE model projections
+  # 3. calculate difference of GDP trends between country and REMIND region based on SSP GDP projections
+  # 4. project energy in the future with the above formulate using the terms calculated in 1. to 3.
+
+  ### define functions ----
+
+  # function get historical energy data for 2020
+  getHistData <- function(subtype, AvgSeveralYears) {
+    # get historical FE and SE data from IEA energy balances
+    IEA <- calcOutput("FE", source = "IEA", ieaVersion = "default", aggregate = F)
+
+    # add items to calculate historical FE demand or historical SE generation
+    if (subtype == "FE") {
+      xHist <- collapseDim(IEA[, , "FE|Buildings (EJ/yr)"] + IEA[, , "FE|Industry (EJ/yr)"] + IEA[, , "FE|Transport (EJ/yr)"], dim = 3)
+    } else if (subtype == "SE|Electricity") {
+      xHist <- collapseDim(IEA[, , "SE|Electricity|Biomass (EJ/yr)"] + IEA[, , "SE|Electricity|Solar (EJ/yr)"] + IEA[, , "SE|Electricity|Wind (EJ/yr)"] +
+                             IEA[, , "SE|Electricity|Hydro (EJ/yr)"] + IEA[, , "SE|Electricity|Geothermal (EJ/yr)"] + IEA[, , "SE|Electricity|Coal (EJ/yr)"] + IEA[, , "SE|Electricity|Gas (EJ/yr)"] + IEA[, , "SE|Electricity|Oil (EJ/yr)"] +
+                             IEA[, , "SE|Electricity|Nuclear (EJ/yr)"], dim = 3)
+    }
+
+    if (AvgSeveralYears) {
+      AvgYears <- c("y2018", "y2019", "y2020", "y2021", "y2022")
+      xHist <- collapseDim(dimSums(xHist[, AvgYears, ], dim = 2) / length(AvgYears))
+    } else {
+      # only get 2020 year
+      xHist <- collapseDim(xHist[, "y2020", ])
+    }
+
+    return(xHist)
+  }
+
+  # function to calculate FE trend on level of REMIND regions based on EDGE data
+  calcFeTrend <- function(subtype, subset, years) {
```

Comment:
I would not call it like this, as it could be confused with a madrat calc function, which it is not.